src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.Xml.cs

```diff
@@ -0,0 +1,204 @@
+                                break;
+
+                            case "G":
+                                // G is always present
+                                reader.Read();
+                                if (string.IsNullOrEmpty(reader.Value))
+                                    ThrowInvalidXmlException("G");
+
+                                dsaParams.G = Convert.FromBase64String(Helpers.DiscardWhiteSpaces(reader.Value));
+                                break;
+
+                            case "Y":
+                                // Y is always present
+                                reader.Read();
+                                if (string.IsNullOrEmpty(reader.Value))
+                                    ThrowInvalidXmlException("Y");
+
+                                dsaParams.Y = Convert.FromBase64String(Helpers.DiscardWhiteSpaces(reader.Value));
+                                break;
+
+                            case "J":
+                                // J is optional
+                                reader.Read();
+                                if (!string.IsNullOrEmpty(reader.Value))
+                                {
+                                    dsaParams.J = Convert.FromBase64String(Helpers.DiscardWhiteSpaces(reader.Value));
+                                }
+                                break;
+
+                            case "X":
+                                // X is optional -- private key
+                                reader.Read();
+                                if (!string.IsNullOrEmpty(reader.Value))
+                                {
+                                    dsaParams.X = Convert.FromBase64String(Helpers.DiscardWhiteSpaces(reader.Value));
+                                }
+                                break;
+
+                            case "Seed":
+                                reader.Read();
+                                if (!string.IsNullOrEmpty(reader.Value))
+                                {
+                                    seed = reader.Value;
+                                }
+
+                                break;
+
+                            case "PgenCounter":
+                                reader.Read();
+                                if (!string.IsNullOrEmpty(reader.Value))
+                                {
+                                    pgenCounter = reader.Value;
+                                }
+                                break;
+                        }
+                    }
+                }
+            }
+
+            // Seed and PgenCounter are optional as a unit -- both present or both absent
+            if ((seed != null) && (pgenCounter != null))
+            {
+                dsaParams.Seed = Convert.FromBase64String(Helpers.DiscardWhiteSpaces(seed));
+                dsaParams.Counter = Helpers.FromBigEndian(Convert.FromBase64String(Helpers.DiscardWhiteSpaces(pgenCounter)));
```

Comment:
Included with new test for seed/counter. Note there was an existing test that generates parameters and round-trips them (at least under Windows anyway - Unix does not create Seed\Counter) so that indirectly tested